hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -63,19 +68,42 @@
     private final HazelcastClientInstanceImpl client;
     private final ILogger logger;
     private final AtomicReference<PartitionTable> partitionTable =
-            new AtomicReference<PartitionTable>(new PartitionTable(null, -1, new Int2ObjectHashMap<Address>()));
+            new AtomicReference<>(new PartitionTable(null, -1, new Int2ObjectHashMap<>()));
     private volatile int partitionCount;
     private volatile long lastCorrelationId = -1;
+    private final CopyOnWriteArrayList<PartitionLostListener> partitionLostListeners = new CopyOnWriteArrayList<>();
 
     public ClientPartitionServiceImpl(HazelcastClientInstanceImpl client) {
         this.client = client;
         this.logger = client.getLoggingService().getLogger(ClientPartitionService.class);
         clientExecutionService = (ClientExecutionServiceImpl) client.getClientExecutionService();
+
+        ClassLoader classLoader = client.getClientConfig().getClassLoader();
```

Comment:
Could we move this code to `start()` to avoid tracking listeners in `partitionLostListeners`?